### PR TITLE
Switch to using gem_install macro

### PR DESCRIPTION
**Pull request by ehelms**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.14
* [ ] 1.13
* [ ] 1.12

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: rubygem-fog/rubygem-fog.spec [hunk 1]
```diff
@@ -86,14 +86,10 @@ Summary:    Documentation for rubygem-%{gem_name}
...
-%{?scl:scl enable %{scl} "}
-gem unpack %{SOURCE0}
-%{?scl:"}
 %setup -q -D -T -n  %{gem_name}-%{version}
-
-%{?scl:scl enable %{scl} "}
-gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
-%{?scl:"}
+%{?scl:scl enable %{scl} - <<EOF}
+%gem_install -n %{SOURCE0}
```
Comment (domcleal) [inline rubygem-fog/rubygem-fog.spec]: This macro is in the wrong location.

File: rubygem-fog/rubygem-fog.spec [hunk 2]
```diff
@@ -86,14 +86,10 @@ Summary:    Documentation for rubygem-%{gem_name}
 This package contains documentation for rubygem-%{gem_name}.
 
 %prep
-%{?scl:scl enable %{scl} "}
-gem unpack %{SOURCE0}
```
Comment (domcleal) [inline rubygem-fog/rubygem-fog.spec]: This style should not be removed

File: rubygem-gettext/rubygem-gettext.spec [hunk 3]
```diff
@@ -61,43 +61,11 @@ This package contains documentation for %{pkg_name}.
...
-cd %{gem_name}-%{version}
-
-#Patches, etc
-
-%{?scl:scl enable %{scl} "}
-gem specification -l --ruby %{SOURCE0}
-%{?scl:"} > %{gem_name}.gemspec
-
-%{?scl:scl enable %{scl} "}
-gem build %{gem_name}.gemspec
```
Comment (domcleal) [inline rubygem-gettext/rubygem-gettext.spec]: Don't remove all of this, it isn't necessary to use gem_install.

File: rubygem-hashie/rubygem-hashie.spec [hunk 4]
```diff
@@ -50,29 +50,12 @@ Summary:   Documentation for %{pkg_name}
 This package contains documentation for %{pkg_name}
 
 %prep
-%{?scl:scl enable %{scl} "}
-gem unpack %{SOURCE0}
```
Comment (domcleal) [inline rubygem-hashie/rubygem-hashie.spec]: Don't remove this.

File: rubygem-openscap/rubygem-openscap.spec [hunk 5]
```diff
@@ -63,29 +63,20 @@ BuildArch: noarch
 Development files for %{name}
 
 %prep
-gem unpack %{SOURCE0}
```
Comment (domcleal) [inline rubygem-openscap/rubygem-openscap.spec]: This shouldn't be removed.

File: rubygem-powerbar/rubygem-powerbar.spec [hunk 6]
```diff
@@ -52,32 +52,12 @@ Summary:   Documentation for %{pkg_name}
 This package contains documentation for %{pkg_name}
 
 %prep
-%{?scl:scl enable %{scl} "}
-gem unpack %{SOURCE0}
```
Comment (domcleal) [inline rubygem-powerbar/rubygem-powerbar.spec]: This shouldn't be removed.

File: rubygem-rabl/rubygem-rabl.spec [hunk 7]
```diff
@@ -36,53 +36,33 @@ Summary: Documentation for rubygem-%{gem_name}
 This package contains documentation for rubygem-%{gem_name}.
 
 %prep
-%{?scl:scl enable %{scl} "}
-gem unpack %{SOURCE0}
```
Comment (domcleal) [inline rubygem-rabl/rubygem-rabl.spec]: This shouldn't be removed.

File: rubygem-rabl/rubygem-rabl.spec [hunk 8]
```diff
@@ -36,53 +36,33 @@ Summary: Documentation for rubygem-%{gem_name}
...
-.travis.yml,rabl.gemspec,test.watchr}
+rm %{buildroot}%{gem_instdir}/{.gitignore,.travis.yml,rabl.gemspec,test.watchr}
 
 %files
 %dir %{gem_instdir}
 %{gem_instdir}/lib
 %{gem_cache}
 %{gem_spec}
-%doc MIT-LICENSE
+%doc %{gem_instdir}/MIT-LICENSE
```
Comment (domcleal) [inline rubygem-rabl/rubygem-rabl.spec]: This isn't correct, the LICENSE should be in the build root if it was unpacked.

File: rubygem-rabl/rubygem-rabl.spec [hunk 9]
```diff
@@ -36,53 +36,33 @@ Summary: Documentation for rubygem-%{gem_name}
...
 
 %files doc
 %doc %{gem_docdir}
 %{gem_instdir}/Gemfile*
 %{gem_instdir}/Rakefile
-%doc README.md CHANGELOG.md CONTRIBUTING.md
+%doc %{gem_instdir}/README.md
+%doc %{gem_instdir}/CHANGELOG.md
+%doc %{gem_instdir}/CONTRIBUTING.md
```
Comment (domcleal) [inline rubygem-rabl/rubygem-rabl.spec]: ditto

File: rubygem-rubyforge/rubygem-rubyforge.spec [hunk 10]
```diff
@@ -87,13 +78,16 @@ popd
...
 %dir %{gem_instdir}
 %{gem_instdir}/lib/
 %{gem_instdir}/bin/
 %{gem_dir}/cache/%{gem_name}-%{version}.gem
 %{gem_dir}/specifications/%{gem_name}-%{version}.gemspec
+%dir %{gem_instdir}/bin
+%{gem_instdir}/bin/rubyforge
+%{_bindir}/rubyforge
 
+%doc %{gem_docdir}
```
Comment (domcleal) [inline rubygem-rubyforge/rubygem-rubyforge.spec]: Must be in a doc subpackage.

File: rubygem-sequel/rubygem-sequel.spec [hunk 11]
```diff
@@ -44,33 +44,21 @@ This package contains documentation for %{pkg_name}.
 
 
 %prep
-%{?scl:scl enable %{scl} "}
-gem unpack %{SOURCE0}
```
Comment (domcleal) [inline rubygem-sequel/rubygem-sequel.spec]: Do not remove this.

File: rubygem-text/rubygem-text.spec [hunk 12]
```diff
@@ -46,37 +46,11 @@ Documentation for %{pkg_name}
 
 %prep
 %setup -n %{pkg_name}-%{version} -q -c -T
-
-TOPDIR=$(pwd)
-mkdir tmpunpackdir
-pushd tmpunpackdir
-
-%{?scl:scl enable %{scl} - << \EOF}
-gem unpack -V %{SOURCE0}
```
Comment (domcleal) [inline rubygem-text/rubygem-text.spec]: Do not remove this.

File: rubygem-wicked/rubygem-wicked.spec [hunk 13]
```diff
@@ -44,29 +44,12 @@ Summary:   Documentation for %{pkg_name}
 This package contains documentation for %{pkg_name}
 
 %prep
-%{?scl:scl enable %{scl} "}
-gem unpack %{SOURCE0}
```
Comment (domcleal) [inline rubygem-wicked/rubygem-wicked.spec]: Do not remove this.

**Comment by ehelms** [comment]
Thanks for the review @domcleal I got a bit over zealous with adding in the macro. I've updated each spec based on your comments.

**Comment by ehelms** [comment]
Removed the duplicate gem_docdir entry.

**Comment by ehelms** [comment]
The failure looks unrelated to the actual changes.

**Comment by domcleal** [comment]
Merged, thanks @ehelms. I will not rebuild these until required, there's little point in doing so.

**Pull request merged**